### DISCUSSION TITLE
Deploying to Production

### DISCUSSION THREAD

**Discussion by juanzgc** [saleor/saleor#8977] [Deployment] [answered]
I _somewhat_ understand the [Architecture of Saleor](https://docs.saleor.io/docs/3.0/developer/getting-started/architecture) but not entirely as you can tell from this post.

I plan to use Saleor in Production - host the UI (Dashboard and React Storefront) on Vercel and _the rest_ on AWS.

But my real question is _what is the rest?_
What exactly are the other services that I need to host in order to run Saleor in production.

Currently I have Saleor set up locally with Saleor-Platform (but using React Storefront) -> https://github.com/saleor/react-storefront/discussions/87#discussioncomment-1869016

I really appreciate any help in understanding the structure and any recommendations you may have if I wish to use AWS or GCP.
For e.g.) Database? Saleor Core? GraphQL? Any and every requirement to self-host this successfully in the Cloud and have a running instance.

Thanks again!

**Comment by karolkielecki** [comment, marked as answer]
Hi @juanzgc.
So, Saleor's preferred deployment method is Docker. In order to run on production, you need to handle: db (PostgresQL), redis for celery, you can also use redis for Django-cache, opentracing (optinal).

Dashboard and Storefront are static apps--they don't require additional servers so that you can use, e.g., S3 or Vercel. Some of our community members host everything on Heroku.

**Reply by ramzizelfani** [reply]
Hi there, is there any additional config files required to deploy dashboard and storefront on vercel (like setting up a vercel.json file)? I am asking that because I am encountering an error while trying to deploy dashboard to vercel ( Error: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings. Learn More: https://vercel.link/missing-public-directory )
Thank you for your support

**Reply by krzysztofwolski** [reply]
I don't think we have ready use config for the dashboard right now. 
Dashboard builds into `/build` directory. You can use our CI config for s3 deployment as a reference:
https://github.com/saleor/saleor-dashboard/blob/main/.github/workflows/deploy-stable-staging.yaml#L13

Don't forget to set up environment variables with API URI, and probably in your case `APP_MOUNT_URI` should be changed to `/`.

**Reply by ramzizelfani** [reply]
Thank you for your answer. It turned out that vercel doesn't detect the dashboard app as a CRA by default that is why it generates the mentioned error above after the build is complete. All what was needed is to setup the app as CRA  in the project settings and the issue was resolved.

**Comment by juanzgc** [comment]
Hi @karolkielecki ,
Thank you for the quick response! I would love to create a blog post with a guide to deploying Saleor to production. Any possible way to possibly speak with you more in depth on best practices when deploying Saleor to production? 

I believe this would greatly help other developers that don't exactly know where to begin.

**Reply by karolkielecki** [reply]
You can reach me out via PM on Gitter: https://gitter.im/mirumee/saleor

**Reply by lack-of-gravitas** [reply]
@juanzgc did you end up writing the guide --- Saleor documentation is severely lacking as far as deployment to Production is concerned apart from "yes we have dockerized the app and change some ENVs" -- would appreciate some direction on this.

**Comment by deepakgupta-unthinkable** [comment]
Hi @krzysztofwolski is saleor-platform via docker not ready for production use? Found this note on GitHub link "**_Keep in mind this repository is for local development only and is not meant to be deployed in any production environment!_**"
https://github.com/saleor/saleor-platform

**Reply by karolkielecki** [reply]
Hi @deepakgupta-unthinkable, Dockerfile inside Saleor is ready for production. You just need to build and deploy the image. More info here: https://docs.saleor.io/docs/3.x/setup/docker-images

But the `saleor-platform` repo has been designed only for development purposes, not production, as stated.

If you want to deploy Saleor to production, please use the following repos:
- https://github.com/saleor/saleor
- https://github.com/saleor/react-storefront
- https://github.com/saleor/saleor-dashboard

Have you considered using our Cloud? You can create [a free developer account](https://cloud.saleor.io/signup).

**Reply by amityweb** [reply]
But the installation docs of Saleor,  https://docs.saleor.io/docs/3.x/setup/docker-compose, linked to from the above saleor/saleor repo, include instructions to install saleor-platform which it says is for local development only here https://github.com/saleor/saleor-platform

Doesn't make sense to me. 

It also fails in the first step:
`docker-compose.yml: `version` is obsolete`

**Comment by a-atalla** [comment]
I have a slightly modified version of saleor-platform, i meant it to be a production ready docker-compose setup [Saleor Stack](https://github.com/a-atalla/saleor-stack)
